addons/account/models/account_move.py

```diff
@@ -1708,11 +1708,26 @@ def _check_balanced(self):
         if not moves:
             return
 
+        postcommit_check_balanced = self.env.context.get('postcommit_check_balanced')
+        if postcommit_check_balanced == []:
+            # Add postcommit hook only once
+            @self.env.cr.postcommit.add
+            def check_balanced():
+                self._do_check_balanced(postcommit_check_balanced)
+                postcommit_check_balanced.clear()
+
+        if postcommit_check_balanced is not None:
+            postcommit_check_balanced += self.ids
```

Comment:
You are mutating a context key?

Reply:
context may contain key `postcommit_check_balanced` which has a list. I update that list

Reply:
Yes, but context is not supposed to be mutated.

Reply:
> Yes, but context is not supposed to be mutated.

Maybe. I don't really other way to fix performance problem in stable

Reply:
Can't you use `self.env.cr.precommit.data` like done here in `addons/mail/models/mail_thread.py`?
https://github.com/odoo/odoo/commit/e8f7cfd00af9a5f549a8723e687385236a4be695